### DISCUSSION TITLE
Security issue since hosting Slinger at home.  Need security advice please.

### DISCUSSION THREAD

**Discussion by GR0DZ1LLA** [GerryDazoo/Slinger#255] [General]
I host a Slinger at home on a Raspberry Pi 3 and do a port forward into my network for external access.  The Slingbox is hosted elsewhere, unimportant here I believe.  The issue is that I'm now getting bombarded on my Slinger.  I'm assuming I'm getting attacked via the port I opened to forward to the Slinger Pi3.  I disabled SSH on the Pi but I don't think that would matter because I forward 8080 and SSH is 22.

I forward FROM ANY, PORT 8080 to an INTERNAL IP.  Clearly I'm asking for attention by doing it this way.

So I ask, what do I do here to stop the attacks?  Is there a better way to handle this?  Is there any risk here?  

I realize there's no security in place here, anyone with my WANIP:8080/slingbox could essentially watch my feed.

Here's a view of my log in my UDM-PRO:
![unifi-slingbox-threats](https://user-images.githubusercontent.com/115742631/200661601-bf235efc-1d98-4b97-8b1a-a38699c38bff.PNG)

**Comment by GerryDazoo** [comment]
What I've found is hackers will port scan most common service ports (22, 80, 443 etc and 8080 unfortunately ) But they very rarely scan the higher port numbers. I changed the external port number to 65432 and the number of bogus connection attempts went down by 2 orders of magnitude.
I also did the same thing for ssh. I use an external port of 22222. Just to make it easier to remember.
You can also change the default URI from "slingbox" to something else. The a hacker would need to know three things the external IP address, external port and your custom URI to watch your stream. 
If you do get port scanned you should see "Hacker Alert" in the logs.

**Reply by theweatherguy** [reply]
Probably the best way to run a secure access to your slingbox server Mac or PC is to set up a VPN **Server** on any machine on your home network and connect to your slingbox_server.py remotely through the VPN.    It will be secure and no one will be able to hack in.   The free open source VPN that's somewaht easy to setup is **WireGuard.**    WireGuard has been thoroughly reviewed by Steve Gibson of "Security Now" and has been given a big thumbs up. 

There are free Wireguard client apps for iOS and iPad OS and I imagine also for android.  Setting up the Wireguard Server on your machine can be done without too much pulling your hair out.   If you have a Mac, you can follow the instructions on the web link below which have worked flawlessly for me for almost two years.  

So you setup the WireGuard **Server** on your home machine and you connect with the free WireGuard **client** app on your phone or iPad or MacBook.

Main Source:  

[](https://barrowclift.me/post/wireguard-server-on-macos)
 

Other links: []( https://blog.scottlowe.org/2021/06/28/using-wireguard-on-mac-via-cli/

https://blog.scottlowe.org/2021/06/22/making-wireguard-from-homebrew-work-on-an-m1-mac/
)

**Reply by theweatherguy** [reply]
Probably the best way to run a secure access to your slingbox server Mac or PC is to set up a VPN **Server** on any machine on your home network and connect to your slingbox_server.py remotely through the VPN.    It will be secure and no one will be able to hack in.   The free open source VPN that's somewaht easy to setup is **WireGuard.**    WireGuard has been thoroughly reviewed by Steve Gibson of "Security Now" and has been given a big thumbs up. 

There are free Wireguard client apps for iOS and iPad OS and I imagine also for android.  Setting up the Wireguard Server on your machine can be done without too much pulling your hair out.   If you have a Mac, you can follow the instructions on the web link below which have worked flawlessly for me for almost two years.  

So you setup the WireGuard **Server** on your home machine and you connect with the free WireGuard **client** app on your phone or iPad or MacBook.

Main Source:  

[](https://barrowclift.me/post/wireguard-server-on-macos)
 

Other links: []( https://blog.scottlowe.org/2021/06/28/using-wireguard-on-mac-via-cli/

https://blog.scottlowe.org/2021/06/22/making-wireguard-from-homebrew-work-on-an-m1-mac/
)

**Reply by socelillyhammer** [reply]
Would a safer way on top of this is to setup the server on raspberrypi then use wireguard as a vpn?  This way 1. you dont have your computer running 24/7 if you want to connect (leaving it open to potential threats) 2. you have raspberry pi, a stand alone computer only running ther server 24/7, and this way if anyone does hack in they only have access to the RPi?  Unless I am wrong - forgive my ignorance here as I am not that tech savvy.

**Reply by tom76017** [reply]
My base router, Asuswrt, supports OpenVPN server.  It's eastly to setup and plenty fast for Slinger.  I don't like the idea of opening the listed ports.  I also run Fail2ban to protect SSH and Apache2 attacts.

`<html>
<body>


Feature | OpenVPN | WireGuard
-- | -- | --
Date Released | May 2001 | September 2019
Encryption | AES, Blowfish, Camellia | ChaCha20, Poly1305
Code Length | Over 70,000 lines | ~4,000 lines
Open Source | Yes | Yes
Security | Strong | Strong
Privacy | Strong | Needs mitigations
Speeds | Moderate | Fast


</body>
</html>`

**Reply by socelillyhammer** [reply]
@tom76017 excuse my ignorance here, but I pay for express vpn.  Are you basically saying that I can setup my router with expressvpn, the dns from expressvpn or from a domain that I have, not open up a port and have more of a secure access when running the server and leaving it on 24/7?  currently i  have to drop my mcafee firewall for the slinger_server so I can watch remtorely.  So like you, im trying to figure out the best way to secure the remote viewing and port

**Reply by Toei79** [reply]
> @tom76017 excuse my ignorance here, but I pay for express vpn. Are you basically saying that I can setup my router with expressvpn, the dns from expressvpn or from a domain that I have, not open up a port and have more of a secure access when running the server and leaving it on 24/7?

you can add vpn to your router at least in asus, i had in the past express vpn too. but i only use one two devices so its overkill nothing happen to me yet anyway.

express vpn let you add to your router so all your whole internet be protected. just few things you need to do on your router. i have asus and its let there.

**Reply by theweatherguy** [reply]
I may be mistaken, but I think you’re confusing a VPN **client** with a VPN **server**.  I’m pretty sure that Express VPN allows you to setup a **client** on your router.  What you want is a VPN SERVER on your local network, whether it be through a dedicated device, raspberry pi, or a computer.  Do checkout setting up Wireguard **server**. It’s fast, free, open source and very secure.  And there are free clients on iPad iPhone and Android

**Reply by Toei79** [reply]
Express vpn let you do fully router in vpn mode  or just use the app.

**Reply by tom76017** [reply]
For clarification, I run OpenVPN as a server on my home router.  For external connections I tunnel with a client.  My Asus router supports this.  With an active tunnel, devices on the LAN can be addressed with the LAN IP address.  I also have other monitoring protection active to block intrusions.

**_Securing Ports_**
The Internet Assigned Numbers Authority (IANA) administrates the port numbers in the range of 0 - 65,535.

When it comes to assigning port numbers for services that are not registered, only port numbers of the so-called dynamic (private) range of 49,152 - 65,535 should be assigned in order to meet minimum security requirements. However, conflicts with already installed applications can occur.

But in practice, numbers in the range from 0-1,023 are protected, and numbers from 1,024 onwards can be used. For example, operating systems must only allow processes with appropriate privileges to open the server ports that are within the given range.

**Reply by theweatherguy** [reply]
Yes, I'm certain that you can run **_OpenVPN_** as a SERVER on your router and tunnel in using a VPN client on your devices.   That's what I'm suggesting with WireGuard, a newer and faster VPN than OpenVPN.   (But both are good.)   

What I have issue with is that with commercial services like ExpressVPN, **_they_** are running the **servers**....you are running just **clients**.   

When running commercial VPNs, while your outbound address becomes  _their server address_, anyone pinging your open port with _their server address_  will be doing the same thing as if you weren't on a VPN.  An open port is an open port.    Having a different IP address with encryption doesn't change anything, especially if you keep the same VPN connection on for days at a time. 

A lot of people get confused by VPN server function vs VPN client function.

**Reply by socelillyhammer** [reply]
Hi guys.  Thank you all for the advices here as Im still learning this stuff.  @theweatherguy so if im understanding this correctly (from the discisson boards, youtube, research etc.)  I run wireguard on the raspberry (not a cloud server - which I have no clue how to setup anyway), and on my phone, computer etc (remote device) I also run wireguard.  my home server will have a encrypted key, when I login via my remote device that will also have an encrypted key and the two devices will talk to each other only via the specific tunnel without hopefully compromising other devices I have on the home network?

**Comment by theweatherguy** [comment]
@socelillyhammer   You pretty much nailed it.  No need to install on a separate Raspberry pi device which is a whole new frontier to learn.
Just install WireGuard Server on your same Windows machine running the Slingbox_server.py  server.  Wireguard has almost no overhead and you won't even know it's running. Incredibly fast, state of the art and probably better than OpenVPN. 
Be sure to follow instructions to install a server configuration, not a client (most web links are about clients.)

Here's one for Windows:
https://www.henrychang.ca/how-to-setup-wireguard-vpn-server-on-windows/


Here's one for MacOS:
https://barrowclift.me/post/wireguard-server-on-macos


This way you don't have to open any ports in your router to the server machine.  When you tunnel in with a VPN, you're essentially on your home local network.  It's like magic! :-)

To find your local machine when you're outside the home, get a throw away free domain name.  https://www.noip.com/   NoiP also has a app that updates their redirect server if you ISP changes your external dynamic address.   

(When you configure the server, use the free domain name, not the numeric external IP address that your ISP dynamically gives you.)

There are free Wireguard clients apps for iPhone, iPad and Android.   Put in the keys and you're in business.

**Reply by socelillyhammer** [reply]
@theweatherguy Hope all is well.  I need some help with Wireguard.  I installed it on RPi but running the server now is shot.  Let me know what I need to post or show you.  Thanks

**Reply by socelillyhammer** [reply]
fixed it.  had to update my port on the config file to my port on the vpn that i set.  it works!!!!! so cool

**Reply by theweatherguy** [reply]
Wireguard is an incredible VPN.  I hope others try it as well.  Much more secure than opening a port and you can do screen sharing and file sharing securely as well.    Did you use the link in my tutorial to install it, or did you find another set of instructions?

**Reply by socelillyhammer** [reply]
It's freaking amazing. I just logged into my home remotely via the VPN, got
into my router and unfowarded the port (closed the port I opened).  Ran the
server with all ports closed and voila, it worked with wireguard.  I have
to learn how to really utilize the vpn but it's amazing.

This was the link I used

https://pimylifeup.com/raspberry-pi-wireguard/

Also a good link if you want to split tunnel the vpn

https://www.wundertech.net/setup-wireguard-on-a-raspberry-pi-vpn-setup-tutorial/

Thanks again for your help

**Reply by socelillyhammer** [reply]
@theweatherguy any luck on installing wireguard on a firestick and getting the VPN QR code installed?  cant find anything on the net

**Reply by theweatherguy** [reply]
Sorry, I haven't installed Wireguard anywhere except on two iMacs.

**Reply by Toei79** [reply]
If you get sone kind of new models router from asus, you get instaguard  , its the same function, like openvpn or wireguard, I guess its easy one its do just activate in the router and download  the app in android or apple.

**Reply by socelillyhammer** [reply]
Thank you. solve a different route.  Downloaded wireguard through their website.  Downloaded a wireless file manager and a file explorer on the stick.  uploaded the config file from the comp via the wireless manager, installed it on wireguard.  Tested it out, it works.  Amazing stuff

**Reply by socelillyhammer** [reply]
Just an update with wireguard.  I found that if your dhcp releases on your router, wg stops working and you'd have to manually deal with reconfiguring everything or just reinstall the Pi OS and everything else.

I found this tool whcih would help keep resolve the dns ip address issue  so wanted to pass this along for anyone using wg or pi

https://techoverflow.net/2021/08/19/how-to-automatically-re-resolve-dns-in-wireguard-on-linux/

and for the non-savvy but curious to learn and understand like myself....this is a great explanation on wg and how it all works

https://sigmdel.ca/michel/ha/wireguard/wireguard_02_en.html

**Comment by skijohn877** [comment]
If you have a Raspberry Pi, have you thought about just setting up UFW? Very easy, and you can block all sorts of bad actors. I have mine limited to a few ip addresses.
